### PR TITLE
Refactor a few client object store components

### PR DESCRIPTION
**Pull request by jmchilton**
A couple fixes but mostly refactoring and modernization related to a couple of the client pieces added in #14073 to make room for user defined object stores in #15875.

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: client/src/components/ObjectStore/ConfigurationMarkdown.vue [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<script setup lang="ts">
+import { computed } from "vue";
+import { markup } from "./configurationMarkdown";
+interface Props {
+    markdown: string;
+    admin: boolean; // was the configuration specified by an admin
+}
+const props = defineProps<Props>();
+const markdownHtml = computed(() => markup(props.markdown || "", props.admin));
```
Comment (davelopez) [inline client/src/components/ObjectStore/ConfigurationMarkdown.vue]:
```suggestion
const markdownHtml = computed(() => markup(props.markdown ?? "", props.admin));
```

Very minor but just to be consistent with the [Styleguide](https://github.com/galaxyproject/galaxy/blob/dev/client/docs/styleguide.md#default-value-assignment) :wink:

File: client/src/components/ObjectStore/ObjectStoreBadge.test.js [hunk 2]
```diff
@@ -39,4 +39,13 @@ describe("ObjectStoreBadge", () => {
         expect(popoverText).toContain(TEST_MESSAGE);
         expect(popoverText).toContain("less secure by the Galaxy administrator");
     });
+
+    it("should gracefully unspecified badge messages", async () => {
```
Comment (davelopez) [inline client/src/components/ObjectStore/ObjectStoreBadge.test.js]:
```suggestion
    it("should gracefully handle unspecified badge messages", async () => {
```

**Review by davelopez — APPROVED** [review]
Looks great!

File: client/src/components/ObjectStore/ConfigurationMarkdown.vue (hunk 1 above)
Comment (jmchilton) [reply client/src/components/ObjectStore/ConfigurationMarkdown.vue]: Very nice - thanks for the link! That is really good to know and I didn't know the difference!

**Comment by jmchilton** [comment]
Thanks for the review and merge!

**Pull request merged**